smack-experimental/src/main/java/org/jivesoftware/smackx/httpfileupload/HttpFileUploadManager.java

```diff
@@ -502,6 +507,46 @@ private void upload(InputStream iStream, long fileSize, Slot slot, UploadProgres
         }
     }
 
+    private static HttpURLConnection createURLConnection(XMPPConnection connection, URL putUrl) throws IOException {
+        Objects.requireNonNull(putUrl);
+        ProxyInfo proxyInfo = fetchProxyInfo(connection);
+        if (proxyInfo != null) {
+            return createProxiedURLConnection(proxyInfo, putUrl);
+        } else {
+            return (HttpURLConnection) putUrl.openConnection();
+        }
+    }
+
+    private static HttpURLConnection createProxiedURLConnection(ProxyInfo proxyInfo, URL putUrl) throws IOException {
+        Objects.requireNonNull(proxyInfo);
+        Objects.requireNonNull(putUrl);
+        final Proxy.Type type = convertProxyType(proxyInfo.getProxyType());
+        final String hostname = proxyInfo.getProxyAddress();
+        final int port = proxyInfo.getProxyPort();
+        final SocketAddress proxySocketAddress = new InetSocketAddress(hostname, port);
+        final Proxy proxy = new Proxy(type, proxySocketAddress);
+        return (HttpURLConnection) putUrl.openConnection(proxy);
+    }
+
+    private static Proxy.Type convertProxyType(ProxyInfo.ProxyType type) {
+        switch (type) {
+            case HTTP:
+                return Proxy.Type.HTTP;
+            case SOCKS4:
+            case SOCKS5:
+                return Proxy.Type.SOCKS;
+        }
+        return Proxy.Type.DIRECT;
```

Comment:
This can never happen, right? So I'd add a `default` case and throw an `AssertionError` in this case. I also think that this method should be added to `ProxyInfo`, e.g. as `ProxyInfo.getJavaNetProxyType()`.

Reply:
Makes sense. Ill make the required changes and push in a bit.